pkg/providers/ovirt/validation/validators/network-mapping-validator.go

```diff
@@ -0,0 +1,154 @@
+	var failures []ValidationFailure
+	// Check whether mapping for network is required and was provided
+	if mapping == nil && len(nics) > 0 {
+		failures = append(failures, ValidationFailure{
+			ID:      NetworkMappingID,
+			Message: "Network mapping is missing",
+		})
+		return failures
+	}
+	// Get all networks needed by the VM as slice of sources
+	requiredNetworks := getRequiredNetworks(nics)
+	mapByID := make(map[string]v2vv1alpha1.ResourceMappingItem)
+	mapByName := make(map[string]v2vv1alpha1.ResourceMappingItem)
+	// Map source id and name to ResourceMappingItem
+	for _, item := range *mapping {
+		if item.Source.ID != nil {
+			mapByID[*item.Source.ID] = item
+		}
+		if item.Source.Name != nil {
+			mapByName[*item.Source.Name] = item
+		}
+	}
+	requiredTargetsSet := make(map[v2vv1alpha1.ObjectIdentifier]*string)
+	// Validate that all vm networks are mapped and populate requiredTargetsSet for target existence check
+	for _, network := range requiredNetworks {
+		if network.ID != nil {
+			item, found := mapByID[*network.ID]
+			if found {
+				requiredTargetsSet[item.Target] = item.Type
+				continue
+			}
+		}
+		if network.Name != nil {
+			item, found := mapByName[*network.Name]
+			if found {
+				requiredTargetsSet[item.Target] = item.Type
+				continue
+			}
+		}
+		failures = append(failures, ValidationFailure{
+			ID:      NetworkMappingID,
+			Message: fmt.Sprintf("Required source network '%s' lacks mapping", ToLoggableID(network.ID, network.Name)),
+		})
+	}
+
+	// Validate that all target networks needed by the VM exist in k8s
+	for networkID, networkType := range requiredTargetsSet {
+		if networkType == nil {
+			continue
+		}
+		if failure, valid := v.validateNetwork(networkID, *networkType, defaultNamespace); !valid {
+			failures = append(failures, failure)
+		}
+	}
+	return failures
+}
+
+func (v *NetworkMappingValidator) validateNetwork(networkID v2vv1alpha1.ObjectIdentifier, networkType string, defaultNamespace string) (ValidationFailure, bool) {
+	switch networkType {
+	case "pod":
+		return ValidationFailure{}, true
+	case "multus":
+		return v.isValidMultusNetwork(networkID, defaultNamespace)
+	default:
+		return ValidationFailure{
+			ID:      NetworkTypeID,
+			Message: fmt.Sprintf("Network %s has unsupported network type: %s", ToLoggableResourceName(networkID.Name, networkID.Namespace), networkType),
+		}, false
+	}
+}
+
+func (v *NetworkMappingValidator) isValidMultusNetwork(networkID v2vv1alpha1.ObjectIdentifier, defaultNamespace string) (ValidationFailure, bool) {
+	namespace := defaultNamespace
+	if networkID.Namespace != nil {
+		namespace = *networkID.Namespace
+	}
+	_, err := v.provider.Find(networkID.Name, namespace)
+	if err != nil {
+		return ValidationFailure{
+			ID:      NetworkTargetID,
+			Message: fmt.Sprintf("Network %s/%s has not been found. Error: %v", networkID.Name, namespace, err),
```

Comment:
It is commonly used to have namespace/name.

Reply:
Good catch!